src/DependencyInjection/Compiler/AutoRegisterCompilerPass.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Compiler;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Sonata\AdminBundle\Annotation\Admin;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class AutoRegisterCompilerPass implements CompilerPassInterface
+{
+    const DEFAULT_SERVICE_PREFIX = 'app.admin.';
+
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->getParameter('sonata.admin.annotations')) {
+            return;
+        }
+
+        $reader = $container->get('annotation_reader');
+
+        foreach ($this->findFiles($container->getParameter('sonata.admin.annotations.directory')) as $file) {
+            if (!($className = $this->getFullyQualifiedClassName($file))) {
+                continue;
+            }
+
```

Comment:
Here you need https://github.com/kunicmarko20/SonataAnnotationBundle/pull/24 this check